### ISSUE TITLE
"WARNING: unrecognised option [--peptidoforms]"

### ISSUE THREAD

**Issue by Shinya-Watanabe** [Nesvilab/FragPipe#1826]
Hi Fengchao,
I wanted to run DIA-NN with "--peptidoforms" function, but it seems like it did not recognized the function.

[log_2024-10-14_10-24-29.txt](https://github.com/user-attachments/files/17367518/log_2024-10-14_10-24-29.txt)

Following #1815, I am still trying to get PTM localization probability using the combination of DIA-NN and FragPipe, which protein c-terminal modification seem the issue not only in PTMProphet but also DIA-NN. I have been discussing this with Vadim on https://github.com/vdemichev/DiaNN/issues/1207. If you have any thoughts on this, it would helpful. 

Best,
Shinya

**Comment by fcyu** [comment]
You need to specify DIA-NN 1.9.1 in FragPipe.

Best,

Fengchao

**Comment by Shinya-Watanabe** [comment]
I got an error `ERROR: D:\diann\src\diann.cpp: 4064: unknown modification: UniMod:734`, which I did not get in the previous version with the same search. 
[log_2024-10-14_12-01-22.txt](https://github.com/user-attachments/files/17367812/log_2024-10-14_12-01-22.txt)

Also, I have unknown modifications for myself,  such as 202.05896 and 216.07462. These are also in library.tsv but I do not have any modifications with these mass in MSFragger. 
`WARNING: below are the unknown modifications found in the library. While unlikely to have a signficant effect, these might negatively affect identification performance in certain scenarious, it is strongly recommended to declare them using any of --mod, --fixed-mod or --var-mod: 202.05896, 216.07462, 87.026746, UniMod:734`

Best,
Shinya

**Comment by fcyu** [comment]
> I got an error ERROR: D:\diann\src\diann.cpp: 4064: unknown modification: UniMod:734, which I did not get in the previous version with the same search.

You need to declare them in DIA-NN https://github.com/vdemichev/DiaNN?tab=readme-ov-file#command-line-reference

> Also, I have unknown modifications for myself, such as 202.05896 and 216.07462. These are also in library.tsv but I do not have any modifications with these mass in MSFragger.

They are the modification mass + amino acid mass.

Best,

Fengchao

**Comment by Shinya-Watanabe** [comment]
I did declare `--var-mod PS,87.03203,DE --var-mod PE,43.0422,DE`. However, DIA-NN does not have function to specify protein c-terminus. Do I need to declare these two as well, 202.05896 and 216.07462? 

This modification `ERROR: D:\diann\src\diann.cpp: 4064: unknown modification: UniMod:734` is already declared by `--var-mod PE,43.0422,DE`.

**Comment by fcyu** [comment]
> I did declare --var-mod PS,87.03203,DE --var-mod PE,43.0422,DE

> This modification ERROR: D:\diann\src\diann.cpp: 4064: unknown modification: UniMod:734 is already declared by --var-mod PE,43.0422,DE.

The mass should be modification mass+amino acid mass if they were not recognized as UniMod 

>  However, DIA-NN does not have function to specify protein c-terminus. Do I need to declare these two as well, 202.05896 and 216.07462?

Then, you won't be able to use the "--peptidoforms" function.

Best,

Fengchao

**Comment by Shinya-Watanabe** [comment]
Why does UniMod:734 has the unknown modification error? This is a modification listed in UniMod, so it should be known modification, isn't it?

Best,
Shinya

**Comment by fcyu** [comment]
It has been well documented https://github.com/vdemichev/DiaNN?tab=readme-ov-file#ptms-and-peptidoforms:~:text=Does%20DIA%2DNN%20need%20to%20recognise%20modifications%20in%20the%20spectral%20library%3F

Best,

Fengchao

**Comment by Shinya-Watanabe** [comment]
I figured this out. The spectral library generated by FragPipe annotates modifications provided in MSFragger as either UniMod accession number or the modification mass. Thus, "cmd line opts" has to use either of them, for example, ` --mod 216.07462,216.07462`. However, I still did not get modification localization columns such as `STY:79.96633 Best Localization` in report.pr_matrix.tsv. When I used "DIA_SpecLib_Quant_Phospho_diaPASEF" workflow, I have those columns. Is this due to disabling PTMProphet?

Best,
Shinya
